enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx

```diff
@@ -63,6 +63,15 @@ export const SdkIframeEmbedSetupProvider = ({
     [settings],
   );
 
+  // Use parameter list hook for dynamic parameter loading
+  const { availableParameters, isLoadingParameters } = useParameterList({
+    experience,
+
+    // We're always using numeric IDs for previews.
+    ...(settings.dashboardId && { dashboardId: Number(settings.dashboardId) }),
```

Comment:
wouldn't it make more sense if `dashboardId` could only be `number` in the first place? Every time I see casting, I'm worried something might break.

Reply:
Well... as for typing, since this is what we control, I'd expect we would only change the type, not the code, if that makes sense. Casting something to a number means it could possibly be something else like a string. But since this is all under our control, I don't even think there's a way for customers to provide string ID (entity ID) for this at all.

Would it make more sense just to cast the type?

```suggestion
    ...(settings.dashboardId && { dashboardId: settings.dashboardId as number }),
```

I'd expect the actual type casting (value casting) to be done if inputs are from customers.

Reply:
Casting just the type makes sense! Fixed in https://github.com/metabase/metabase/pull/60233/commits/2c57ce901ad9fc4e9d8cfa5b90e6a45fd39a592a